relayer-cli/src/commands/tx/client.rs

```diff
@@ -163,3 +164,64 @@ impl Runnable for TxUpgradeClientCmd {
         }
     }
 }
+
+#[derive(Clone, Command, Debug, Options)]
+pub struct TxUpgradeClientsCmd {
+    #[options(free, required, help = "identifier of the chain that must be upgraded")]
+    src_chain_id: ChainId,
+}
+
+impl Runnable for TxUpgradeClientsCmd {
+    fn run(&self) {
+        let config = app_config();
+
+        let outputs: Vec<Output> = config
+            .chains
+            .iter()
+            .map(|chain| match self.upgrade_chain(&config, &chain.id) {
+                Ok(receipts) => Output::success(receipts),
+                Err(e) => Output::error(e),
+            })
+            .collect();
+
+        Output::combined(outputs.into_iter()).exit()
+    }
+}
+
+impl TxUpgradeClientsCmd {
+    fn upgrade_chain(
+        &self,
+        config: &config::Reader<CliApp>,
+        dst_chain_id: &ChainId,
+    ) -> Result<Vec<IbcEvent>, Error> {
+        let chains = ChainHandlePair::spawn(&config, &self.src_chain_id, dst_chain_id)?;
```

Comment:
Note that here we are re-`spawn`-ing the chain runtime for the source chain every time the method `upgrade_chain` is called. Instead of using `ChainHandlePair::spawn` there may be slightly more efficient ways to do it. One alternative, for instance:

- in the caller (`TxUpgradeClientsCmd::run`) we spawn the chain runtime for the source chain a single time, using something like [`spawn_chain_runtime`](https://github.com/informalsystems/ibc-rs/blob/7712422a47c25563e8c211fcbf79fcaabec142ec/relayer-cli/src/cli_utils.rs#L38) which would return the chain handle for the source chain id.
- then we pass this chain handle (or a clone thereof) when calling `self.upgrade_client(&config, &chain_id, &src_chain_handle)`.